src/main/java/IO/DbWriter.java

```diff
@@ -7,9 +7,10 @@
 
 /* DbWriter class provides output stream from database to client.
  * If user makes DbWriter object and calls the write() function,
- * user SHOULD NOT close the Outputstream returned directly. Instead, the user
- * must use the finish() function to handle closing the inputstream and any
- * other necessary end operations.  */
+ * user either close the Outputstream returned directly or
+ * use the finish() function to handle closing the inputstream and any
+ * other necessary end operations, BUT DO NOT CALL BOTH.
```

Comment:
This API will be tricky for a user to use correctly, because they have to find and read this comment, then make the choice without any clear guidelines on how to choose. Always try to streamline APIs to make them harder to misuse, and prefer not to make users obey comments (which they often will miss) except as a last resort.

In this particular case I don't think you need to do anything special at all. Both `InputStream` and `OutputStream` say their `close` method does nothing.

https://docs.oracle.com/javase/8/docs/api/java/io/OutputStream.html#close--
https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html#close--

Reply:
oh wowwww, interesting. will ask more in pod sync